src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivityContext.netcoreapp.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Diagnostics
+{
+    /// <summary>
+    /// ActivityContext representation conforms to the w3c TraceContext specification. It contains two identifiers
+    /// a TraceId and a SpanId - along with a set of common TraceFlags and system-specific TraceState values.
+    /// </summary>
+    public readonly partial struct ActivityContext : IEquatable<ActivityContext>
+    {
+        public override int GetHashCode()
+        {
+            if (this == default)
+                return 0;
+
+            return HashCode.Combine(TraceId, SpanId, TraceFlags, TraceState == null ? "" : TraceState);
```

Comment:
Doesn't HashCode.Combine do null checks?  I'm wondering if the last argument can just be: `TraceState`.  Also, do we expect `this == default` to be common?  If not, should we just remove that and always call Combine?